YakimS/assignment2/BinaryTree.java

```diff
@@ -0,0 +1,231 @@
+package CodeUAss1Package;
+
+import javax.management.openmbean.KeyAlreadyExistsException;

```

Comment:
It is better not to depend on such complex packages just for exception. Better use something from `java.lang`, like `IllegalArgumentException`.